python/ribasim/ribasim/model.py

```diff
@@ -443,11 +577,22 @@ def plot(self, ax=None) -> Any:
         if ax is None:
             _, ax = plt.subplots()
             ax.axis("off")
+
         self.network.edge.plot(ax=ax, zorder=2)
         self.plot_control_listen(ax)
         self.network.node.plot(ax=ax, zorder=3)
 
-        ax.legend(loc="lower left", bbox_to_anchor=(1, 0.5))
+        handles, labels = ax.get_legend_handles_labels()
```

Comment:
This should ideally not be part of the PR (but will ignore it here)?

Reply:
You mean the whole subnetwork plotting thing? I guess it would have bean cleaner if that was done in a separate PR, but it did help with debugging of the model merging